storage/src/cache/fscache/mod.rs

```diff
@@ -148,27 +148,27 @@ impl BlobCacheMgr for FsCacheMgr {
     }
 
     fn gc(&self, id: Option<&str>) {
-        let mut reclaim = Vec::new();
-
         if let Some(blob_id) = id {
-            reclaim.push(blob_id.to_string());
+            self.blobs.write().unwrap().remove(blob_id);
         } else {
+            let mut reclaim = Vec::new();
             let guard = self.blobs.write().unwrap();
             for (id, entry) in guard.iter() {
                 if Arc::strong_count(entry) == 1 {
                     reclaim.push(id.to_owned());
                 }
             }
-        }
-
-        for key in reclaim.iter() {
-            let mut guard = self.blobs.write().unwrap();
-            if let Some(entry) = guard.get(key) {
-                if Arc::strong_count(entry) > 1 {
-                    continue;
+            drop(guard);
+
+            for key in reclaim.iter() {
+                let mut guard = self.blobs.write().unwrap();
+                if let Some(entry) = guard.get(key) {
+                    if id.is_none() && Arc::strong_count(entry) > 1 {
```

Comment:
id should be always none here?

Reply:
My mistake, `id.is_none() &&` should be removed